### ISSUE TITLE
[Feature] search issues when written in commit message

### ISSUE THREAD

**Issue by lingfengyu-dreaming** [sourcegit-scm/sourcegit#702]
After setting the issue rules, you can link to the mentioned issue in commit. Can I search for cited issues in the Submit Message Panel and display links? This allows you to check the issues mentioned before committing.

**Comment by love-linger** [comment]
Do you mean the `autocomplete` feature?

![image](https://github.com/user-attachments/assets/aa93e07c-56b5-4ac8-bfe9-4b04f73b66a9)

**Comment by lingfengyu-dreaming** [comment]
I think it is similar to `autocomplete`.
If not work like this, it can open the issue link before confirm the commit.(Just like the issue in commit history)

**Comment by alensiljak** [comment]
Which issue tracker would it be linked to? This will not work offline.

**Comment by emptylight370** [comment]
Initially, I just wanted a feature that would allow jumping to the corresponding website from the commit panel based on a defined issue tracker, without having to wait for the commit to be submitted to see the jump in the commit history panel. Later, the mention of GitHub's issue auto-completion seemed like an advanced feature.

**Comment by goran-w** [comment]
This idea might be out of scope for SourceGit. It seems to require a deeper integration with specific issue trackers, and parsing of response data requested from web, in order to display a filtered list of titles etc from available / relevant issue-pages on the issue-tracker site. This might require authentication (login) etc, too.

The existing feature of linking to mentioned issues (and PRs etc) is much simpler:
* Look for specified issue-reference pattern (for GitHub, Jira etc) in commit message
* Use extracted issue-reference to assemble a URL
* Open URL in web-browser

**Comment by Mingun** [comment]
Common, the topicstarter requests ability to have a clickable link when writing commit message, and not only in read-only text control that shows commit messages.

**Comment by goran-w** [comment]
> Common, the topicstarter requests ability to have a clickable link when writing commit message, and not only in read-only text control that shows commit messages.

Ah, sorry - now I understand. 😅 The requested feature is to make issue-references clickable in the commit-message editor, while writing the message, so these links can be opened in browser (for checking it's the correct issue, etc).

@love-linger Would this be possible to implement? There's been some recent changes to the commit-message editor (monospace font, width-guide line etc). How much work would it be to add clickable links within the title/body of the message?

**Comment by JC-Chung** [comment]
Maybe we could add a small icon/button in the commit message toolbar that shows the matched issue tracker link(s), so users don't have to wait until the commit is made to jump to the referenced issue.

**Comment by goran-w** [comment]
> Maybe we could add a small icon/button in the commit message toolbar that shows the matched issue tracker link(s), so users don't have to wait until the commit is made to jump to the referenced issue.

Good idea! That might be a simpler way to implement this, instead of inserting clickable links in the text itself (which is being edited "live")?
